### PR TITLE
Added tests for callbacks

### LINKED ISSUE #62: Fix comments in chatpage.js
Remove istanbul ignore comments and increase coverage to acceptable levels

### PR DESCRIPTION
**Pull request by Jordan141**
**Reference to related ticket:**
resolves #62 
**Description of changes:**
Removed hacky comments and added tests for callbacks. Review requested for test quality.

### CODE REVIEW

File: src/client/src/components/chatpage.js [hunk 1]
```diff
@@ -25,18 +25,22 @@ class ChatPage extends React.Component {
...
 	}
 	componentDidMount() {
-		/* istanbul ignore next */
-		socket.on(protocols.CONNECTION, () => this.setState({connected: true}))
-		/* istanbul ignore next */
-		socket.on(protocols.MESSAGE, data => this.updateMessageState(data))
-		/* istanbul ignore next */
-		socket.on(protocols.JOIN, data => this.updateJoinedRooms(data))
-		/* istanbul ignore next */
+		socket.on(protocols.CONNECTION, this.handleSocketData)
```
Comment (michaelKurowski) [inline src/client/src/components/chatpage.js]: Are you sure that you won't lose binding to `this` due to the fact that it's being used by `socket.on` as callback? Same question to other cases.

File: src/client/src/.tests/components/chatpage.test.js [hunk 2]
```diff
@@ -191,5 +191,21 @@ describe('<ChatPage />', () => {
...
+			const CONNECT_STATUS = suite.Component.state(CONNECTED)
+			expect(CONNECT_STATUS).toBeTruthy()
+			expect(RETURN_VALUE).toBeUndefined()
+		})
+		it('Should handle the message event by updating message state with data', () => {
+			const MESSAGE_DATA =  {username: DUMMY_USER, message: DUMMY_MESSAGE, timestamp: DUMMY_TIMESTAMP}
+			const RETURN_VALUE = suite.Component.instance().handleMessageEvent(MESSAGE_DATA)
+			expect(RETURN_VALUE).toBeUndefined()
+		})
+		it('Should handle the join event by updating userrooms state with data', () => {
```
Comment (michaelKurowski) [inline src/client/src/.tests/components/chatpage.test.js]: You don't check here if it updates state

File: src/client/src/.tests/components/chatpage.test.js [hunk 3]
```diff
@@ -191,5 +191,21 @@ describe('<ChatPage />', () => {
 			suite.Component.instance().sendMessage()
 			expect(suite.Component.state(MESSAGES).length).toBe(EXPECTED_ELEMENTS_COUNT)
 		})
+		it('Should handle the connection event by setting connection state to true', () => {
+			const RETURN_VALUE = suite.Component.instance().handleConnectionEvent()
+			const CONNECT_STATUS = suite.Component.state(CONNECTED)
+			expect(CONNECT_STATUS).toBeTruthy()
+			expect(RETURN_VALUE).toBeUndefined()
+		})
+		it('Should handle the message event by updating message state with data', () => {
```
Comment (michaelKurowski) [inline src/client/src/.tests/components/chatpage.test.js]: You don't check here if it updates state

File: src/client/src/.tests/components/chatpage.test.js [hunk 4]
```diff
@@ -191,5 +191,21 @@ describe('<ChatPage />', () => {
 			suite.Component.instance().sendMessage()
 			expect(suite.Component.state(MESSAGES).length).toBe(EXPECTED_ELEMENTS_COUNT)
 		})
+		it('Should handle the connection event by setting connection state to true', () => {
```
Comment (michaelKurowski) [inline src/client/src/.tests/components/chatpage.test.js]: You don't check here if it updates state

File: src/client/src/components/chatpage.js (hunk 1 above)
Comment (Jordan141) [reply src/client/src/components/chatpage.js]: Confirmed, https://pastebin.com/aDAewCXf for details

File: src/client/src/.tests/components/chatpage.test.js (hunk 2 above)
Comment (Jordan141) [reply src/client/src/.tests/components/chatpage.test.js]: Corrected

File: src/client/src/.tests/components/chatpage.test.js (hunk 3 above)
Comment (Jordan141) [reply src/client/src/.tests/components/chatpage.test.js]: Corrected

**Pull request merged**